js/components/admin-page.js

```diff
@@ -3709,17 +3696,23 @@ class AdminPage {
     }
 
     updateDashboardDisplay() {
-        // Add demo data indicator if using fallback values
-        if (this.contractStats.isDemo) {
-            this.addDemoDataIndicator();
+        // Don't update display if stats failed to load (error already shown)
+        if (!this.contractStats) {
+            return;
         }
 
-        // Update stat values
+        // Update stat values - show N/A for null, actual values for available data
         const elements = {
-            'active-pairs-count': this.contractStats.activePairs || 0,
-            'total-tvl': this.formatNumber(this.contractStats.totalTVL || 0),
-            'total-stakers': this.contractStats.totalStakers || 0,
-            'total-rewards': this.formatNumber(this.contractStats.totalRewards || 0)
+            'active-pairs-count': this.contractStats.activePairs ?? 0,
+            'total-tvl': this.contractStats.totalTVL !== null 
+                ? `$${this.formatNumber(this.contractStats.totalTVL)}` 
+                : 'N/A',
+            'total-stakers': this.contractStats.totalStakers !== null 
+                ? this.contractStats.totalStakers 
+                : 'N/A',
+            'total-rewards': this.contractStats.totalRewards !== null 
+                ? this.formatNumber(this.contractStats.totalRewards) 
+                : 'N/A'
```

Comment:
```suggestion
            'active-pairs-count': this.contractStats.activePairs ?? 0,
            'total-tvl': this.contractStats.totalTVL != null 
                ? `$${this.formatNumber(this.contractStats.totalTVL)}` 
                : 'N/A',
            'total-stakers': this.contractStats.totalStakers != null 
                ? this.contractStats.totalStakers 
                : 'N/A',
            'total-rewards': this.contractStats.totalRewards != null 
                ? this.formatNumber(this.contractStats.totalRewards) 
                : 'N/A'
```
change double equal to single equal to catch undefined